zcash_primitives/src/note_encryption.rs

```diff
@@ -527,7 +574,7 @@ pub fn try_sapling_output_recovery(
     }
 
     let to = PaymentAddress::from_parts(diversifier, pk_d)?;
-    let note = to.create_note(v, rcm, &JUBJUB).unwrap();
+    let note = to.create_note(v, rseed, &JUBJUB).unwrap();
 
     if note.cm(&JUBJUB) != *cmu {
```

Comment:
After this `cmu` check, add the [missing ZIP 212 `esk` check](https://github.com/zcash/zcash/blob/67ab09ae43caf17c0331b5c7a0aa26e08ee252b2/src/zcash/Note.cpp#L398-L404) (using the new function on `Note`:
```rust
    if let Rseed::AfterZip212(rseed) = note.rseed {
        if note.derive_esk().unwrap() != esk {
            return None;
        }
    }
```